jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/HTTP3Stream.java

```diff
@@ -60,6 +61,27 @@ public HTTP3Stream(HTTP3Session session, QuicStreamEndPoint endPoint, boolean lo
         this.local = local;
     }
 
+    public Callback cancel(Throwable cause, Callback callback)
+    {
+        Callback nested = new Callback.Nested(callback)
+        {
+            @Override
+            public void succeeded()
+            {
+                super.failed(cause);
+            }
+
+            @Override
+            public void failed(Throwable x)
+            {
+                ExceptionUtil.addSuppressedIfNotAssociated(cause, x);
+                super.failed(cause);
+            }
+        };
+        reset(HTTP3ErrorCode.REQUEST_CANCELLED_ERROR.code(), cause);
+        return nested;
+    }
```

Comment:
@lorban that looks simple enough.  Same as h2.  Any idea how to test this?

Reply:
This is the big question: the implementation looks fairly easy but I've been scratching my head trying to find a way to reliably test that.